electricitymap/contrib/lib/models/events.py

```diff
@@ -119,6 +137,30 @@ def set_value(
             self._corrected_negative_values.add(mode)
         self.__setattr__(mode, value)
 
+    def add_value(
```

Comment:
I see a potential edge case here with negative values.

Imagine that we have the existing_value set to 2 and we get a new value with -1.
In this case we would call self.set_value with 2+(-1) = 1. Therefore we don't trigger the correct negative values function, which I think we should do. Also in this way we are mixing the self consumption from one mode with another which I don't think is super clear. But I acknowledge that how we always did it, so not sure it's that big of an issue.

Reply:
In my opinion it would make more sense to correct for each endpoint added individually before adding them together.

Reply:
Added a internal function to correct each endpoint, I think it might make sense to also record the amount for each mode that was corrected, mostly for accounting purposes in the future.

Reply:
Let's maybe not overcomplexify this as we are not using the amount yet. So let's just focus on one change and not take into account the amount.